pkg/internal/managedappcredential/manager.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedappcredential
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	controllerconfig "github.com/gardener/gardener-extension-provider-openstack/pkg/apis/config"
+	openstackclient "github.com/gardener/gardener-extension-provider-openstack/pkg/openstack/client"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewManager returns a new manager to manage the lifecycle of
+// the managed appplication credentials of an Openstack Shoot cluster.
+func NewManager(openstackClientFactory openstackclient.FactoryFactory, config *controllerconfig.ApplicationCredentialConfig, client client.Client, namespace, identifier, finalizerName string, logger logr.Logger) *Manager {
+	return &Manager{
+		client:                 client,
+		config:                 config,
+		finalizerName:          finalizerName,
+		identifier:             identifier,
+		logger:                 logger,
+		namespace:              namespace,
+		openstackClientFactory: openstackClientFactory,
+	}
+}
+
+func (m *Manager) runGarbageCollection(ctx context.Context, parent *parent, inUseAppCredentialID string, deleteInUseAppCredential bool) error {
```

Comment:
From the uses in `actuator_delete` and `actuator_reconcile` it is not clear to me if we need both inUseAppCredentialID and deleteInUseAppCredential. The general idea is to not delete the current in-use things and also have a mode where you delete everything. Isn't the same if you pass an empty in-use credential to delete everything ?

Reply:
No we don't need both. It would be sufficient to pass the `appCredentialID` as a `*string` and check for nil. That's how I had it in the beginning and then changed to current version for better readability. Now I'm rather with you and changed is back.